apps/snitch_core/test/data/schema/payment/card_payment_test.exs

```diff
@@ -13,18 +26,98 @@ defmodule Snitch.Data.Schema.CardPaymentTest do
 
   describe "CardPayment records" do
     test "refer card type Payments (uniquely)", context do
-      %{ccd: card} = context
-      card_payment = CardPayment.changeset(%CardPayment{}, %{payment_id: card.id}, :create)
+      %{ccd: card_payment, user: user} = context
+
+      card_payment =
+        CardPayment.changeset(
+          %CardPayment{},
+          %{payment_id: card_payment.id, card: Map.put(@card, :user_id, user.id)},
+          :create
+        )
+
       assert {:error, %{errors: errors}} = Repo.insert(card_payment)
       assert errors == [payment_id: {"has already been taken", []}]
     end
 
     test "DON'T refer other Payments", context do
-      %{chk: check} = context
-      card_payment = CardPayment.changeset(%CardPayment{}, %{payment_id: check.id}, :create)
-      Repo.insert(card_payment)
+      %{chk: check, user: user} = context
+
+      card_payment =
+        CardPayment.changeset(
+          %CardPayment{},
+          %{payment_id: check.id, card: Map.put(@card, :user_id, user.id)},
+          :create
+        )
+
       assert {:error, %Ecto.Changeset{errors: errors}} = Repo.insert(card_payment)
       assert errors == [payment_id: {"does not refer a card payment", []}]
     end
+
+    test "when added new card", context do
```

Comment:
This test will not cover the `else` block of `assoc_card`, and this execution flow path is already _covered_ in tests above. IMO, this test is pointless.